### PR TITLE
feat: make Observability Operator version used by KFM configurable by CLI flags

### PR DESCRIPTION
**Pull request by vmanley**
## Description

Jira: [MGDSTRM-9499](https://issues.redhat.com/browse/MGDSTRM-9499), [MGDSTRM-8588](https://issues.redhat.com/browse/MGDSTRM-8588)
This PR will allow for the Observability Operator version to be provided via a flag.
Also the default version of the Observability Operator deployed by KFM is upgraded from v3.0.10 to v3.0.14
## Verification Steps

* From this PR, KFM can be run locally with the flags:
  * `--observability-operator-index-image=quay.io/rhoas/observability-operator-index:v3.0.10`
  * `--observability-operator-starting-csv=observability-operator.v3.0.10`
This will install v3.0.10 of the Observability Operator. Allow the Operator to fully install then stop KFM running locally.

* Run KFM again with flags for upgrade index image and starting CSV values:
  * `--observability-operator-index-image=quay.io/rhoas/observability-operator-index:v3.0.14`
  * `--observability-operator-starting-csv=observability-operator.v3.0.14`
This will upgrade the existing Observability Operator from v3.0.10 to v3.0.14. Upgrade should proceed without error.

**Note:** Currently KFM deploys v3.0.10 of the Observability Operator. This change will install v3.0.14 by default. Upgrade behaviour across multiple versions has been check as part of [MGDSTRM-9499](https://issues.redhat.com/browse/MGDSTRM-9499)
## Checklist (Definition of Done)

- [x] All acceptance criteria specified in JIRA have been completed
- [x] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [x] Documentation added for the feature
- [x] CI and all relevant tests are passing
- [x] Code Review completed
- [x] Verified independently by reviewer
- [x] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] ~~Required metrics/dashboards/alerts have been added (or PR created).~~
- [ ] ~~Required Standard Operating Procedure (SOP) is added.~~
- [ ] ~~JIRA has been created for changes required on the client side~~

### CODE REVIEW

**Review by machi1990 — CHANGES_REQUESTED** [review]
The change looks good to me but I wondering maybe we've to evaluate doing https://issues.redhat.com/browse/MGDSTRM-8588.
Please see https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1035#discussion_r876119610

**Comment by pb82** [comment]
@machi1990 agree. How about including the additional flag in this PR? We also need to update the service template and prepare an app-interface MR.

**Comment by vmanley** [comment]
Thanks @machi1990. Completing [MGDSTRM-8588](https://issues.redhat.com/browse/MGDSTRM-8588) would be a benefit. A colleague volunteered to attempt this task but seems like they weren't able to complete it

**Comment by vmanley** [comment]
@pb82 @machi1990 I can include the change in this PR

**Comment by machi1990** [comment]
+1 on including the flags in this PR @pb82 @vmanley . 

Thanks @vmanley for taking on the task.

**Comment by MikeEdgar** [comment]
> A colleague volunteered to attempt this task but seems like they weren't able to complete it

@vmanley maybe this we me? I started down the path of making Observability subscription config configuration the same way its done for Strimzi and kas-fleetshard. It's a bit out of date now with conflicts that I haven't had a chance to resolve. Is that something the team would be interested in including (versus a simple param for the version)?

**Comment by vmanley** [comment]
@MikeEdgar It wasn't yourself I had in mind but thanks for having a look at this. I did start off this change looking at how Strimzi and kas-fleetshard operators are installed but I'm not sure it's necessary/available for Observability Operator. I'm working on just configuring the index image used by KAS Fleet Manager to install Obs Operator

File: Makefile
Comment (machi1990) [inline Makefile:776]: Can we update this doc section as well https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/4d81751ebd17a92ad43a918106b37e58ba6d8e05/docs/deploying-kas-fleet-manager-to-openshift.md#deploy-kas-fleet-manager?

**Review by machi1990 — COMMENTED** [review]
the change lgtm, thanks @vmanley 

I've left one doc update suggestion

**Comment by miguelsorianod** [comment]
Will the observability operator be upgraded seamlessly in already existing data planes?

File: templates/service-template.yml
Comment (machi1990) [inline templates/service-template.yml:551]: Once this is merged, can we update app-interface to include the new parameters?

**Comment by miguelsorianod** [comment]
Changes look good to me

File: templates/service-template.yml
Comment (vmanley) [reply templates/service-template.yml:551]: New default version (v3.0.14) should be picked and Operator should upgrade. I can update app-interface after this is merged

**Comment by vmanley** [comment]
> Will the observability operator be upgraded seamlessly in already existing data planes?

@miguelsorianod Multi-version upgrade was checked and progressed fine. Process should only take a couple of minutes

**Review by miguelsorianod — APPROVED** [review]

**Comment by miguelsorianod** [comment]
:+1: . Can be merged once tests pass

**Review by machi1990 — APPROVED** [review]

**Comment by vmanley** [comment]
Follow-up MR for app-interface: https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/47265

**Pull request merged**
